neuromation/api/storage.py

```diff
@@ -357,6 +370,7 @@ def _uri_to_path(self, uri: URL) -> str:
                         f"Cannot upload {child}, not regular file/directory",
                     )
                 )  # pragma: no cover
+        await asyncio.gather(*tasks)
```

Comment:
Here there is asyncio quirk: `gather` doesn't cancel inner tasks if a future returned by `gather` call is canceled.

As you don't use results returned by `await gather()` the `gather()` is sub-optimal here I think.
Please consider the following instead:
```
try:
    done, tasks = await asyncio.wait(tasks, return_when=asyncio.FIRST_EXCEPTION)
    for task in done:
        await task
except BaseException:
    for task in tasks:
        task.cancel()
    await asyncio.wait(tasks)  # wait for actual cancellation, ignore all exceptions raised from tasks
```

What do you think about?

Reply:
Good catch, thanks!

This quirk is even specially highlighted in the documentation.